bcda/service/service_test.go

```diff
@@ -1170,6 +1186,206 @@ func (s *ServiceTestSuite) TestGetACOConfigForID() {
 	}
 }
 
+func (s *ServiceTestSuite) TestACOConfigurations_Integration() {
+}
+
+type ServiceTestSuiteWithDatabase struct {
+	suite.Suite
+	priorityACOsEnvVar string
+	repository         *postgres.Repository
+	db                 *sql.DB
+}
+
+func TestServiceTestSuiteWithDatabase(t *testing.T) {
+	suite.Run(t, new(ServiceTestSuiteWithDatabase))
+}
+
+func (s *ServiceTestSuiteWithDatabase) SetupSuite() {
+	s.db, _ = databasetest.CreateDatabase(s.T(), "../../db/migrations/bcda/", true)
+	s.repository = postgres.NewRepository(s.db)
+}
+
+func (s *ServiceTestSuiteWithDatabase) SetupTest() {
+	s.priorityACOsEnvVar = conf.GetEnv("PRIORITY_ACO_REG_EX")
+}
+
+func (s *ServiceTestSuiteWithDatabase) TearDownTest() {
+	conf.SetEnv(s.T(), "PRIORITY_ACO_REG_EX", s.priorityACOsEnvVar)
+}
+
+// suppressions.yml
+// MBI00000001: (opted out) opted out 1 day ago and has an older record with no preference
+// MBI00000002: (opted out) opted out 10 days ago and opted in 30 days ago
+// MBI00000003: (opted out) has no previous records and opted out 1 day ago
+// MBI00000004: has a single record with no opt out preference ("")
+// MBI00000005: has no previous record but has opted in
+// MBI00000006: has previous opt out records and opted in
+// MBI00000007: has no opt out records
+// MBIs 1,2,3 should be suppressed for ACOs that have the ignore_suppressions set in config
+func (s *ServiceTestSuiteWithDatabase) TestGetBenesByID_Integration() {
+	cfg, err := LoadConfig()
+	if err != nil {
+		s.T().Fatal(err)
+	}
+
+	service := service{
+		repository: s.repository,
+		sp: suppressionParameters{
+			includeSuppressedBeneficiaries: false,
+			lookbackDays:                   60,
+		},
+		acoConfigs: cfg.ACOConfigs,
+	}
+
+	test_cases := []struct {
+		name      string
+		cmsID     string
+		beneCount int
+		mbis      []string
+	}{
+		{"ACO config ignore suppressions is true", "CT000001", 7, []string{"MBI00000001", "MBI00000002", "MBI00000003", "MBI00000004", "MBI00000005", "MBI00000006", "MBI00000007"}},
+		{"ACO config ignore suppressions is false", "A0001", 4, []string{"MBI00000004", "MBI00000005", "MBI00000006", "MBI00000007"}},
```

Comment:
Looks like we are testing the first two but would like to see the third as well:
- ACO config with ignoreSuppressions = true
- ACO config with ignoreSuppressions = false
- ACO config with no mention of ignoreSuppressions (should default to false)

Reply:
Good point - I will do an explicit set to false, since the last bullet point is applied to all the other acos in the config.